### DISCUSSION TITLE
MicroPython ESP32 - Manipulation of multiple pins by reading and writing "registers"

### DISCUSSION THREAD

**Discussion by pdr0663** [micropython/micropython#10893] [ESP32]
Hi, I'm interested in creating an 8-bit ADC using a resistor ladder to demonstrate audio output. Is it possible to write output pins as a group by writing to a "register", as is possible on Arduino Unos and Nanos? I'm planning to write at 4kHz, so I suspect that calling the python routines for each single pin would be too slow.

Paul

**Comment by 2dof** [comment]
You can use viper code emitter (for speed)   to write  byte directly to  port output memory register.
see:  https://forum.micropython.org/viewtopic.php?t=7162
       
 -> technical ref. esp32 ( to find out registers adress) : https://www.mouser.com/pdfdocs/ESP32-Tech_Reference.pdf         
( section 4.12 Register Summary)
